Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -5418,42 +5691,133 @@ public override void AdvancedAdhesion(float elapsedClockSeconds)
+
+                LocomotiveAxles[0].DriveType = AxleDriveType.ForceDriven;
+                LocomotiveAxles[0].BrakeRetardForceN = BrakeRetardForceN;
+                LocomotiveAxles[0].DriveForceN = SteamEngines[numberofengine].TractiveForceN;              //Total force applied to wheels
+
+                LocomotiveAxles[0].TrainSpeedMpS = SpeedMpS;               //Set the train speed of the axle mod
+                LocomotiveAxles[0].WheelRadiusM = SteamEngines[numberofengine].DriveWheelRadiusM;
+                LocomotiveAxles[0].AxleSpeedMpS = SteamEngines[numberofengine].previousAxleSpeedMpS;
+
+                LocomotiveAxles[0].Update(elapsedClockSeconds); //Main updater of the axle model
+
+                SteamEngines[numberofengine].previousAxleSpeedMpS = LocomotiveAxles[0].AxleSpeedMpS;
+                SteamEngines[numberofengine].CompensatedAxleForceN = LocomotiveAxles[0].CompensatedAxleForceN;
+
+                //                Trace.TraceInformation("AxleForce - EngNum# {0} DriveForce {1} CompAxle {2} AxleCompAxle {3} Friction {4} Inertia {5} Damping {6}", numberofengine, LocomotiveAxle.DriveForceN, SteamEngines[numberofengine].CompensatedAxleForceN, LocomotiveAxle.CompensatedAxleForceN, LocomotiveAxle.FrictionN, LocomotiveAxle.InertiaKgm2, LocomotiveAxle.DampingNs);
+
+
+                if (elapsedClockSeconds > 0)
+                {
+                    WheelSlip = LocomotiveAxles[0].IsWheelSlip;             //Get the wheelslip indicator
+                    WheelSlipWarning = LocomotiveAxles[0].IsWheelSlipWarning && SlipControlSystem != SlipControlType.Full;
+                    SteamEngines[numberofengine].WheelSlip = LocomotiveAxles[0].IsWheelSlip;
+                }
+
+                if (AbsSpeedMpS <= 0.15 && !WheelSlip)
+                {
+                    WheelSpeedSlipMpS = SpeedMpS;
+                    WheelSpeedMpS = SpeedMpS;
+                }
+                else
+                {
+                    WheelSpeedSlipMpS = LocomotiveAxles[0].AxleSpeedMpS;
+                    WheelSpeedMpS = SpeedMpS;
+                }
+
+
+
+
+
+
+
+
+
+            }
+
+        }
+
+
+        /// <summary>
+        /// This function applies a sign to the motive force as a function of the direction of the train.
+        /// </summary>
+        protected virtual void ApplyDirectionToMotiveForce()
+        {
+            if (Train.IsPlayerDriven)
+            {
+                switch (Direction)
+                {
+                    case Direction.Forward:
+                        //MotiveForceN *= 1;     //Not necessary
+                        break;
+                    case Direction.Reverse:
+                        MotiveForceN *= -1;
+                        break;
+                    case Direction.N:
+                    default:
+                        MotiveForceN *= 0;
+                        break;
+                }
             }
-            base.AdvancedAdhesion(elapsedClockSeconds);
```

Comment:
We absolutely need to call AdvancedAdhesion from the base class. Ideally the overriden method should be as small as possible.